c10/core/TensorImpl.h

```diff
@@ -2435,10 +2435,24 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return is_non_overlapping_and_dense_default();
   }
 
+  // if this returns true, then it is guaranteed that this tensor has symbolic
+  // sizes/strides
   bool has_symbolic_sizes_strides() const {
     return has_symbolic_sizes_strides_;
   }
 
+  // if this returns true, then it is guaranteed that this tensor does NOT have
+  // symbolic sizes/strides. This is different from the above, because it's
+  // possible that has_symbolic_sizes_strides() returns false, but we do
+  // not have symbolic sizes/strides. This exists for the case of
+  // Nested Tensor python subclass, where the sizes are implemented in python
+  // (TODO: clean this up and just implement sizes in nested tensor without a
```

Comment:
The naming still feels a little confusing to me since `has_symbolic_sizes_strides() != !does_not_have_symbolic_sizes_strides()` when the naming suggests that they should be inverses?

If we wanted to do something temporary to block NT (e.g. we were planning to remove this) and avoid renaming all callsites of has_symbolic_sizes_strides, maybe its just easiest to directly modify the callsite in `execution_trace_observer.cpp` to also check `matches_policy(SizesStridesPolicy::CustomStrides)`?

Reply:
sorry, didn't see this after it was edited I think. I think `matches_policy` is `protected` so it's not accessible outside of `TensorImpl`. I guess we could also make that public, but I'm not sure if that's too different from this change?